x-pack/plugins/inference/common/create_output_api.test.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { firstValueFrom, isObservable, of, toArray } from 'rxjs';
+import {
+  ChatCompleteResponse,
+  ChatCompletionEvent,
+  ChatCompletionEventType,
+} from '@kbn/inference-common';
+import { createOutputApi } from './create_output_api';
+
+describe('createOutputApi', () => {
+  let chatComplete: jest.Mock;
+
+  beforeEach(() => {
+    chatComplete = jest.fn();
+  });
+
+  it('calls `chatComplete` with the right parameters', async () => {
+    chatComplete.mockResolvedValue(Promise.resolve({ content: 'content', toolCalls: [] }));
+
+    const output = createOutputApi(chatComplete);
+
+    await output({
+      id: 'id',
+      stream: false,
+      functionCalling: 'native',
+      connectorId: '.my-connector',
+      system: 'system',
+      input: 'input message',
+    });
+
+    expect(chatComplete).toHaveBeenCalledTimes(1);
+    expect(chatComplete).toHaveBeenCalledWith({
+      connectorId: '.my-connector',
+      functionCalling: 'native',
+      stream: false,
+      system: 'system',
+      messages: [
+        {
+          content: 'input message',
+          role: 'user',
+        },
+      ],
+    });
+  });
+
+  it('returns the expected value when stream=false', async () => {
+    const chatCompleteResponse: ChatCompleteResponse = {
+      content: 'content',
+      toolCalls: [{ toolCallId: 'a', function: { name: 'foo', arguments: { arg: 1 } } }],
+    };
+
+    chatComplete.mockResolvedValue(Promise.resolve(chatCompleteResponse));
+
+    const output = createOutputApi(chatComplete);
+
+    const response = await output({
+      id: 'my-id',
+      stream: false,
+      connectorId: '.my-connector',
+      input: 'input message',
+    });
+
+    expect(response).toEqual({
+      id: 'my-id',
+      content: chatCompleteResponse.content,
+      output: chatCompleteResponse.toolCalls[0].function.arguments,
```

Comment:
I can see that this is testing existing code that wasn't materially changed. For my own education, can you explain why this is the expected output for this call?

Reply:
The `output` function is a"forced tool execution" wrapper. When using it, the consumer knows the LLM will call a tool with the provided schema. 

In that test, we're just making sure that the output API's response matches what was returned from the chatComplete API, basically (so it's `content` and the arguments if its first/sole tool call).

Not sure if that answers the question 😄 ?

Reply:
Yes that's helpful - thanks!